### ISSUE TITLE
Add analyzer to detect when a class has more than one `ImportingConstructor` attribute

### ISSUE THREAD

**Issue by zivkan** [microsoft/vs-mef#287]
A class I was modifying was long and the existing constructor wasn't at the top of the file, so I added another. I assume the error was written to the ComponentModelCache/Microsoft.VisualStudio.Default.err file, but neither compiling nor running the experimental instance gave me any indication about this error. I was just lucky that CI tests hit a code path where another component tried to import this class.

Anyway, if the analyzer checked for classes that have more than one `[ImportingConstructor]`, this type of error would be caught more quickly.

**Comment by AArnott** [comment]
Agreed. We should first ensure that discovery on the part produces the error in the log for guaranteed discoverability without the help of analyzers and an open source project.
An analyzer beyond that to prevent the bug in the first place is goodness as well.
